ios/AppLovinMAXAdView.m

```diff
@@ -0,0 +1,204 @@
+    
+    if ( self.adView )
+    {
+        [self.adView setExtraParameterForKey: @"adaptive_banner" value: adaptiveBannerEnabled ? @"true" : @"false"];
+    }
+}
+
+- (void)setAutoRefresh:(BOOL)autoRefresh
+{
+    _autoRefresh = autoRefresh;
+    
+    if ( self.adView )
+    {
+        if ( autoRefresh )
+        {
+            [self.adView startAutoRefresh];
+        }
+        else
+        {
+            [self.adView stopAutoRefresh];
+        }
+    }
+}
+
+- (void)attachAdView
+{
+    if ( ![self.adUnitId al_isValidString] )
+    {
+        [[AppLovinMAX shared] log: @"Attempting to attach MAAdView without Ad Unit ID"];
+        return;
+    }
+    
+    if ( !self.adFormat )
+    {
+        [[AppLovinMAX shared] log: @"Attempting to attach MAAdView without ad format"];
+        return;
+    }
+    
+    // Re-assign in case of race condition
+    NSString *adUnitId = self.adUnitId;
+    MAAdFormat *adFormat = self.adFormat;
+    
+    // Run after 0.25 sec delay to allow all properties to set
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t) (NSEC_PER_SEC/4)), dispatch_get_main_queue(), ^{
+        if ( self.adView )
+        {
+            [[AppLovinMAX shared] log: @"Attempting to re-attach with existing MAAdView: %@", self.adView];
+            return;
+        }
+        [[AppLovinMAX shared] log: @"Attaching MAAdView..."];
+        self.adView = [[MAAdView alloc] initWithAdUnitIdentifier: adUnitId
+                                                        adFormat: adFormat
+                                                             sdk: AppLovinMAX.shared.sdk];
+        self.adView.frame = (CGRect) { CGPointZero, adFormat.size };
+        self.adView.delegate = AppLovinMAX.shared; // Go through core class for callback forwarding to React Native layer
+        self.adView.revenueDelegate = AppLovinMAX.shared;
+        if ( self.placement )
+        {
+            self.adView.placement = self.placement;
+        }
+        if ( self.customData )
+        {
+            self.adView.customData = self.customData;
+        }
+
+        if ( [self isAdaptiveBannerEnabled] )
+        {
+            [self.adView setExtraParameterForKey: @"adaptive_banner" value: @"true"];
+        }
+        else
+        {
+            [self.adView setExtraParameterForKey: @"adaptive_banner" value: @"false"];
+        }
+        
+        // Set this extra parameter to work around a SDK bug that ignores calls to stopAutoRefresh()
+        [self.adView setExtraParameterForKey: @"allow_pause_auto_refresh_immediately" value: @"true"];
+        
+        if ( [self isAutoRefresh] )
+        {
+            [self.adView startAutoRefresh];
+        }
+        else
+        {
+            [self.adView stopAutoRefresh];
+        }
+        
+        [self.adView loadAd];
+        
+        [self addSubview: self.adView];
+        
+        CGSize adViewSize = [self.adFormat adaptiveSizeForWidth: CGRectGetWidth(self.frame)];
```

Comment:
If adaptive banners are disabled - we don't need to get the adaptive size.

Reply:
```
  const sizeForAdFormat = () => {
    if (props.adFormat === AdFormat.BANNER) {
      let width = AppLovinMAX.isTablet() ? 728 : 320;
      let height;

      if (props.adaptiveBannerEnabled) {
        height = AppLovinMAX.getAdaptiveBannerHeightForWidth(-1);
      } else {
        height = AppLovinMAX.isTablet() ? 90 : 50;
      }

      return { width: width, height: height }
    } else {
      return { width: 300, height: 250 }
    }
  };

```

Reply:
Javascript already defines the banner size.   
```
  const sizeForAdFormat = () => {
    if (props.adFormat === AdFormat.BANNER) {
      let width = AppLovinMAX.isTablet() ? 728 : 320;
      let height;

      if (props.adaptiveBannerEnabled) {
        height = AppLovinMAX.getAdaptiveBannerHeightForWidth(-1);
      } else {
        height = AppLovinMAX.isTablet() ? 90 : 50;
      }

      return { width: width, height: height }
    } else {
      return { width: 300, height: 250 }
    }
  };
```
So, I think that we just need to follow this size for both adaptive and non-adaptive banners.